### PR TITLE
feat: use new ID-based followups API

### PR DESCRIPTION
**Pull request by mediremi**
Closes https://jira.dhis2.org/browse/DHIS2-11062

### CODE REVIEW

File: src/pages/FollowUpAnalysis/FollowUpAnalysisTable/ElementRow.js [hunk 1]
```diff
@@ -17,13 +17,13 @@ const ElementRow = ({ element, onCheckboxToggle, onShowComment }) => (
         <TableRowColumn>{element.organisation}</TableRowColumn>
         <TableRowColumn>{element.period}</TableRowColumn>
         <TableRowColumn className={cssPageStyles.numericalRow}>
-            <FormattedNumber value={element.min} />
+            {element.min && <FormattedNumber value={element.min} />}
```
Comment (mediremi) [inline src/pages/FollowUpAnalysis/FollowUpAnalysisTable/ElementRow.js:20]: The new API does not set `min` and `max` to 0 if they are not defined.

Comment (ismay) [reply src/pages/FollowUpAnalysis/FollowUpAnalysisTable/ElementRow.js:20]: Is it possible for element.min to be `0`? Because in that case this would hide the FormattedNumber input. Maybe we should make the check more explicit here if that's the case.

File: src/pages/FollowUpAnalysis/FollowUpAnalysis.js [hunk 2]
```diff
@@ -61,12 +62,15 @@ const FollowUpAnalysis = ({ sectionKey }) => {
...
+    const {
+        loading: loadingFollowUpList,
+        refetch: fetchFollowUpList,
+    } = useDataQuery(query, {
+        lazy: true,
+        onComplete: data => {
+            const elements = data.followups.followupValues.map(
+                convertElementFromApiResponse
+            )
             setElements(elements)
```
Comment (ismay) [inline src/pages/FollowUpAnalysis/FollowUpAnalysis.js:74]: We should get rid of this pattern. But I suppose that's not part of this PR. Would be good to make sure there's an issue for this though.

File: src/pages/FollowUpAnalysis/FollowUpAnalysisTable/ElementRow.js [hunk 3]
```diff
@@ -17,13 +17,13 @@ const ElementRow = ({ element, onCheckboxToggle, onShowComment }) => (
...
         </TableRowColumn>
         <TableRowColumn className={cssPageStyles.numericalRow}>
             <FormattedNumber value={element.value} />
         </TableRowColumn>
         <TableRowColumn className={cssPageStyles.numericalRow}>
-            <FormattedNumber value={element.max} />
+            {element.max && <FormattedNumber value={element.max} />}
```
Comment (ismay) [inline src/pages/FollowUpAnalysis/FollowUpAnalysisTable/ElementRow.js:26]: Same question as above comment.

**Review by ismay — COMMENTED** [review]
Runs well for me locally. Let me know what you think of the comments I left. In general two other things stood out to me:

- This app has no tests it seems, neither unit nor integration. So that means that currently each deploy is somewhat of a yolo deploy. I don't know if there's already an issue for this, but we should check if there is and create it if necessary.
- It seems the org unit tree on the follow ups route has dynamic queries. We should probably move that to either static queries, or use the engine directly. If there's not yet an issue for it we should create one.

Other than the above and the comments it looks good to me.

File: src/pages/FollowUpAnalysis/FollowUpAnalysisTable/ElementRow.js (hunk 1 above)
Comment (mediremi) [reply src/pages/FollowUpAnalysis/FollowUpAnalysisTable/ElementRow.js:20]: I don't know if `element.min` can be 0 but I'll change this check to `typeof element.min === 'number' && ...` since that's both more explicit and safer. (and I'll update the `element.max` check too)

File: src/pages/FollowUpAnalysis/FollowUpAnalysis.js (hunk 2 above)
Comment (mediremi) [reply src/pages/FollowUpAnalysis/FollowUpAnalysis.js:74]: :100: - I'll create a jira issue and make a PR to remove it this pattern from the various components that use it once I've written some cypress tests

Comment (mediremi) [reply src/pages/FollowUpAnalysis/FollowUpAnalysis.js]: Jira issue: https://jira.dhis2.org/browse/DHIS2-11521

**Comment by mediremi** [comment]
> This app has no tests it seems, neither unit nor integration. So that means that currently each deploy is somewhat of a yolo deploy. I don't know if there's already an issue for this, but we should check if there is and create it if necessary.

I've filed https://jira.dhis2.org/browse/DHIS2-11520 and will ensure tests of some kind are implemented before doing further work on this app.

> It seems the org unit tree on the follow ups route has dynamic queries. We should probably move that to either static queries, or use the engine directly. If there's not yet an issue for it we should create one.

I've filed https://jira.dhis2.org/browse/LIBS-201

**Comment by ismay** [comment]
> I've filed https://jira.dhis2.org/browse/DHIS2-11520 and will ensure tests of some kind are implemented before doing further work on this app.
>
> I've filed https://jira.dhis2.org/browse/LIBS-201

Awesome, thanks! By the way, even though I personally think it's good to prioritize adding tests, don't take my opinion on it as leading. In the end our prioritized jira queue is still leading I think. But it'd be good if we move the test issue up in that queue, for sure.

File: src/pages/FollowUpAnalysis/FollowUpAnalysis.js (hunk 2 above)
Comment (ismay) [reply src/pages/FollowUpAnalysis/FollowUpAnalysis.js]: Nice, thanks!

**Review by ismay — APPROVED** [review]
Nice, looks good! 👍
